pkg/appender/appender.go

```diff
@@ -277,29 +273,55 @@ func (mc *MetricsCache) Add(lset utils.LabelsIfc, t int64, v interface{}) (uint6
 		if metric.isVariant {
 			existingValueType = "string"
 		}
-		return 0, errors.Errorf("Cannot append %v type metric to %v type metric.", newValueType, existingValueType)
+		return nil, errors.Errorf("Cannot append %v type metric to %v type metric.", newValueType, existingValueType)
 	}
 
 	mc.appendTV(metric, t, v)
 	for _, aggrMetric := range aggrMetrics {
 		mc.appendTV(aggrMetric, t, v)
 	}
 
-	return metric.refID, err
+	return &MetricIdentifier{metric.name, metric.hash}, err
 }
 
-// fast Add to metric (by refID)
-func (mc *MetricsCache) AddFast(ref uint64, t int64, v interface{}) error {
+func (mc *MetricsCache) AddFast(lset utils.Labels, ref *MetricIdentifier, t int64, v interface{}) error {
 
 	err := verifyTimeValid(t)
 	if err != nil {
 		return err
 	}
+	if ref == nil {
+		return fmt.Errorf("nil identifier")
```

Comment:
You can avoid this by making the parameter `MetricIdentifier` rather than `*MetricIdentifier`. I promise you that using a pointer isn't helping performance in this case 🙂

Reply:
i prefer a pointer so i could return nil on errors instead of empty MetricIdentifier, looks nicer
any reason not to use a pointer?

Reply:
Ah, I see. But here it's a parameter, not a return value :)
Only reason here is to avoid a bad argument to begin with.